packages/manager/src/features/CloudPulse/Alerts/CreateAlert/utilities.ts

```diff
@@ -15,18 +15,48 @@ export const filterFormValues = (
     'region',
     'engineType',
     'severity',
+    'rule_criteria',
   ]);
   // severity has a need for null in the form for edge-cases, so null-checking and returning it as an appropriate type
   const severity = formValues.severity!;
   const entityIds = formValues.entity_ids;
-  return { ...values, entity_ids: entityIds, severity };
+  const rules = formValues.rule_criteria.rules;
+  return {
+    ...values,
+    entity_ids: entityIds,
+    rule_criteria: { rules: filterMetricCriteriaFormValues(rules) },
+    severity,
+  };
 };
 
 export const filterMetricCriteriaFormValues = (
-  formValues: MetricCriteriaForm
+  formValues: MetricCriteriaForm[]
 ): MetricCriteria[] => {
-  const aggregationType = formValues.aggregation_type!;
-  const operator = formValues.operator!;
-  const values = omitProps(formValues, ['aggregation_type', 'operator']);
-  return [{ ...values, aggregation_type: aggregationType, operator }];
+  return formValues.map((rule) => {
+    const values = omitProps(rule, ['aggregation_type', 'operator', 'metric']);
+    return {
+      ...values,
+      aggregation_type: rule.aggregation_type!,
+      dimension_filters: rule.dimension_filters,
+      metric: rule.metric!,
+      operator: rule.operator!,
```

Comment:
Is there a way to avoid this non-null assertion?

Reply:
The metric value is null in the first place just to handle clear option in the Autocomplete, and we have validation to make sure it's a required field and the submit won't happen successfully unless and until the field has the proper value . So the non-null assertion is almost never going to be checked.

We actually did the same thing for Severity.
https://github.com/linode/manager/blob/681acc64856770366b729a7842dc48e08764e9ae/packages/manager/src/features/CloudPulse/Alerts/CreateAlert/utilities.ts#L20

Reply:
It's important to note this isn't a null *check*, it's just a null assertion which will assume the value is non-null even if it isn't. I prefer to avoid it if at all possible unless the team believes it's harmless. For example, we can use a default instead:

```suggestion
      metric: rule.metric ?? '',
      operator: rule.operator ?? 'eq',
```

Reply:
I guess to fix this we'd have to update the form types to not allow things like `metric` and `operator` to be nullable. This might require us to make those fields not clearable to satisfy the types

Reply:
The fields not being clearable should be fine, but for components to be controlled we have initialized with null. I think the solution suggested by @hkhalil-akamai might be more suited for our use-case. I will try that change and if it's not affecting anything I'll make that change and push the changes.

Reply:
@hkhalil-akamai , going ahead with above suggestion.

Reply:
Tested with having the default values like how @hkhalil-akamai  suggested. Will be going ahead with that.